src/h3/api/basic_int/__init__.py

```diff
@@ -395,14 +395,14 @@ def uncompact_cells(cells, res):
     return _out_collection(hu)
 
 
-def h3shape_to_cells(h3shape, res):
+def h3shape_to_cells(shape, res):
```

Comment:
FWIW - I don't want to second-guess this, but I'm not sure how much this approach offers over just offering `polygon_to_cells` and making the caller break multi-polygons into polygons. The caller *still* has to handle this, by choosing `LatLngPoly` or `LatLngMultiPoly` when they convert their input data, but in this context you have to handle the both shapes in separate clauses and lose the clarity of sticking to the core library API. I'm not sure that trade-off is worthwhile for a slight ergonomic improvement when prepping input.

That said, I'm sure you've thought through this, so I'm okay with this API if you think it's worthwhile.

Reply:
I was imagining this being helpful in the API because everything is symmetric and there are no special cases. We have three classes of objects: "geo objects", `H3Shape`s, and cells. For any two types, there's a function that translates between them, and every function has an obvious inverse. To me, that's a nice complete set of functionality, and there are no gotchas like "oh, right... for that case, I need to give it a list of polygons".

And I don't think it is true that the caller has to handle it by choosing either `LatLngPoly` or `LatLngMultiPoly`. I think the typical use case is actually that they call `geo_to_h3shape()` on a GeoPandas column (which are often a mix of Shapely Polygons and MultiPolygons). So think this aligns more with what Python folks expect from working with geopandas.

Also, we have a MultiPolygon class in the core API (its just kind of hard to use), so I think this actually still matches the C code.